bson/src/main/org/bson/codecs/pojo/PojoCodecImpl.java

```diff
@@ -50,36 +50,28 @@
     private final CodecRegistry registry;
     private final PropertyCodecRegistry propertyCodecRegistry;
     private final DiscriminatorLookup discriminatorLookup;
-    private final boolean specialized;
 
     PojoCodecImpl(final ClassModel<T> classModel, final CodecRegistry codecRegistry,
-                  final List<PropertyCodecProvider> propertyCodecProviders, final DiscriminatorLookup discriminatorLookup) {
+            final List<PropertyCodecProvider> propertyCodecProviders, final DiscriminatorLookup discriminatorLookup) {
         this.classModel = classModel;
         this.registry = codecRegistry;
         this.discriminatorLookup = discriminatorLookup;
         this.propertyCodecRegistry = new PropertyCodecRegistryImpl(this, registry, propertyCodecProviders);
-        this.specialized = shouldSpecialize(classModel);
         specialize();
     }
 
-    PojoCodecImpl(final ClassModel<T> classModel, final CodecRegistry codecRegistry, final PropertyCodecRegistry propertyCodecRegistry,
-                  final DiscriminatorLookup discriminatorLookup, final boolean specialized) {
+    PojoCodecImpl(final ClassModel<T> classModel, final CodecRegistry codecRegistry,
+            final PropertyCodecRegistry propertyCodecRegistry, final DiscriminatorLookup discriminatorLookup) {
         this.classModel = classModel;
         this.registry = codecRegistry;
         this.discriminatorLookup = discriminatorLookup;
         this.propertyCodecRegistry = propertyCodecRegistry;
-        this.specialized = specialized;
         specialize();
```

Comment:
Unless we want to duplicate a bunch of code, this can only be done if one of the constructors reuses the other one, which [does not seem possible](https://github.com/mongodb/mongo-java-driver/pull/1136#discussion_r1222003413).